File: CPP/dynamicObject/p3.cpp
#include<iostream>
using namespace std;


class Demo{

	public:
		int x = 10;


};


class Test{

	public:
	

		int x = 2;
		
		Test(){
			cout<<"Constructor"<<endl;

		}

		Test(int a){
			x =a;
		}

	
		
		Test(Demo& ref){
			cout<<ref.x<<endl;
		}
		
/*

		Test(Demo ref){

			cout<<ref.x<<endl;

		}
*/
		

		void display(Demo& ref){
			int a = 10;
			cout<<ref.x<<endl;

		}

		void fun(){
			cout<<x<<endl;
		}

};



int main(){
	Demo obj1;


	Test t2{obj1};
	

	t2.fun();



}

File: java/innerClass/annonymous/p3.java
class X{
	void m1(){
		System.out.println("in m1");
	}
}


class Test{
	public static void main(String args[]){
		X x = new X(){
			public final static int val = 10;
			 void m2(){
				System.out.println("in m2");
			}
			
			 void m1(){

				 class Demo{
				 int b = 50;

			}


				System.out.println(" statement in overridden m1");
				System.out.println(val);
				Demo d = new Demo();
				System.out.println(d.b);


 			
			}	

			
		};

		x.m1();
	//	System.out.println(x.val);




	}
}
